### PR TITLE
PWN-5197 - show shares does not match on error on triggerSignInNoTorus

### PR DESCRIPTION
**Pull request by kabramovp2p**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5197

## Description of Work

Show shares does not match on error on triggerSignInNoTorus

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 1]
```diff
@@ -119,7 +119,9 @@ class UserRestoreInteractor(
         if (web3AuthError.errorType == Web3AuthErrorResponse.ErrorType.CANNOT_RECONSTRUCT) {
             RestoreUserResult.UserNotFound
         } else {
-            RestoreUserResult.RestoreFailed(Throwable("Unknown error type"))
+            RestoreUserResult.SharesDoesNotMatch
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:122]: maybe we can specify which shares are no match? or it's for every restore case?

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 2]
```diff
@@ -93,6 +95,23 @@ class OnboardingGeneralErrorFragment :
                     isVisible = true
                 }
             }
+            is GeneralErrorScreenError.SharesDoesNotMatchError -> {
+                errorState.titleResId?.let { textViewErrorTitle.text = getString(it) }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:99]: textViewErrorTitle.text = getString(it)
->
textViewErrorTitle.setText(it)

File: app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt [hunk 3]
```diff
@@ -93,6 +95,23 @@ class OnboardingGeneralErrorFragment :
...
+                errorState.messageResId?.let { textViewErrorSubtitle.text = getString(it) }
+
+                with(buttonRestoreByGoogle) {
+                    text = getString(R.string.onboarding_general_error_bug_report_button_title)
+                    setOnClickListener { IntercomService.showMessenger() }
+                    setIconResource(R.drawable.ic_caution)
+                    setIconTintResource(R.color.text_night)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/auth/ui/generalerror/OnboardingGeneralErrorFragment.kt:106]: icons_night

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -61,6 +61,8 @@
     <string name="error_wallet_not_found_title">No wallet found</string>
     <string name="error_wallet_not_found_message">Try account or use an another phone number</string>
     <string name="error_too_often_otp_requests_message">You request SMS too often. Try again in %1d sec.</string>
+    <string name="error_shares_does_not_matches_title">Shares aren\'t\nmatch</string>
+    <string name="error_shares_does_not_matches_message">So sorry, baby\n\nIf you will write us, use error code #10464</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/strings.xml:65]: error code should be passed from the web3auth

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt (hunk 1 above)
Comment (kabramovp2p) [reply app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:122]: I presume this is for every case when shares does not match and we can't do anything

File: app/src/main/res/values/strings.xml (hunk 4 above)
Comment (kabramovp2p) [reply app/src/main/res/values/strings.xml:65]: you sure ??? damn

File: app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt [hunk 5]
```diff
@@ -119,7 +119,9 @@ class UserRestoreInteractor(
         if (web3AuthError.errorType == Web3AuthErrorResponse.ErrorType.CANNOT_RECONSTRUCT) {
             RestoreUserResult.UserNotFound
         } else {
-            RestoreUserResult.RestoreFailed(Throwable("Unknown error type"))
+            RestoreUserResult.SharesDoesNotMatch(web3AuthError.errorCode)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/auth/interactor/restore/UserRestoreInteractor.kt:122]: `SharesDoNotMatch`

File: app/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -61,6 +61,8 @@
     <string name="error_wallet_not_found_title">No wallet found</string>
     <string name="error_wallet_not_found_message">Try account or use an another phone number</string>
     <string name="error_too_often_otp_requests_message">You request SMS too often. Try again in %1d sec.</string>
+    <string name="error_shares_does_not_matches_title">Shares aren\'t\nmatch</string>
```
Comment (1sakov) [inline app/src/main/res/values/strings.xml:64]: Incorrect translation. 
Should be `Shares aren't matching` or `Shares do not match`.
Please talk to our product manager about this translation

File: app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -61,6 +61,8 @@
...
+    <string name="error_shares_does_not_matches_message">So sorry, baby\n\nIf you will write us, use error code #%1d</string>
```
Comment (1sakov) [inline app/src/main/res/values/strings.xml:65]: Really? Baby?? Are you sure it's correct?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (kabramovp2p) [reply app/src/main/res/values/strings.xml:65]: @1sakov do you see what i see?
image

**Review by davran1001 — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 7 above)
Comment (kabramovp2p) [reply app/src/main/res/values/strings.xml:65]: fixed with Nina

**Review by gslevinkov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

**Pull request merged**